sklearn/cluster/optics_.py

```diff
@@ -580,13 +586,10 @@ def _automatic_cluster(reachability_plot, ordering,
     """
 
     min_neighborhood_size = 2
-    min_cluster_size = int(min_cluster_size_ratio * len(ordering))
+    if min_cluster_size <= 1:
+        min_cluster_size = max(2, int(min_cluster_size * len(ordering)))
     neighborhood_size = int(min_maxima_ratio * len(ordering))
 
-    # Should this check for < min_samples? Should this be public?
```

Comment:
I don't think it should be min_samples. The minimum cluster size in dbscan is less than min_samples. But I don't see why it  should be 5 either.

Reply:
I have suggested minimum value of 2 in my PR, and changing the name to min_neighbors or something.

Reply:
Yes, I'm accepting your min cluster size. Not so much the min_samples name. See there.